### PR TITLE
update git ignore

### PR DESCRIPTION
**Pull request by MikolajMichaluk**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,4 +3,44 @@ cmake-build-debug/
...
+config.sub
+configure
+depcomp
+example/basic_plots/.deps/
+example/basic_plots/.libs/
+example/basic_plots/amtrack_plot
+example/demo/.deps/
+example/demo/.libs/
+example/demo/amtrack_demo
+example/demo/data.txt
```
Comment (grzanka) [inline .gitignore:32]: What are `data.txt` and `data1.txt` ?

**Comment by grzanka** [comment]
Please add also entries (including comments) from official template:
https://github.com/github/gitignore/blob/master/C.gitignore

File: .gitignore [hunk 2]
```diff
@@ -3,4 +3,40 @@ cmake-build-debug/
...
+*Makefile.in
+INSTALL
+README
+aclocal.m4
+autom4te.cache/
+compile
+config.guess
+config.h
+config.h.in
+config.h.in~
```
Comment (grzanka) [inline .gitignore:28]: Replace with `*~`

File: .gitignore [hunk 3]
```diff
@@ -3,4 +3,40 @@ cmake-build-debug/
...
+*.lo
+*.la 
+
+*.deps/
+*.libs/
+
+*Makefile
+*Makefile.in
+INSTALL
+README
```
Comment (grzanka) [inline .gitignore:21]: Why INSTALL and README ?

File: .gitignore [hunk 4]
```diff
@@ -3,4 +3,40 @@ cmake-build-debug/
...
+
+#libraries
+*.lo
+*.la 
+
+*.deps/
+*.libs/
+
+*Makefile
+*Makefile.in
```
Comment (grzanka) [inline .gitignore:19]: Check if all `*in` files should stay

File: .gitignore [hunk 5]
```diff
@@ -3,4 +3,40 @@ cmake-build-debug/
...
+INSTALL
+README
+aclocal.m4
+autom4te.cache/
+compile
+config.guess
+config.h
```
Comment (grzanka) [inline .gitignore:26]: Check if `config.h` is autogenerated

Comment (MikolajMichaluk) [reply .gitignore:26]: yes, it is

File: .gitignore (hunk 4 above)
Comment (MikolajMichaluk) [reply .gitignore:19]: these files are autogenerated by makefile.am

File: .gitignore (hunk 3 above)
Comment (MikolajMichaluk) [reply .gitignore:21]: that was mistake

File: .gitignore (hunk 2 above)
Comment (MikolajMichaluk) [reply .gitignore]: Done

**Review by grzanka — APPROVED** [review]
Look ok

**Pull request merged**